gstools/transform/array.py

```diff
@@ -0,0 +1,351 @@
+    "array_to_uniform",
+    "array_to_arcsin",
+    "array_to_uquad",
+]
+
+
+def array_discrete(
+    field, values, thresholds="arithmetic", mean=None, var=None
+):
+    """
+    Discrete transformation.
+
+    After this transformation, the field has only `len(values)` discrete
+    values.
+
+    Parameters
+    ----------
+    field : :class:`numpy.ndarray`
+        Normal distributed values.
+    values : :any:`numpy.ndarray`
+        The discrete values the field will take
+    thresholds : :class:`str` or :any:`numpy.ndarray`, optional
+        the thresholds, where the value classes are separated
+        possible values are:
+        * "arithmetic": the mean of the 2 neighbouring values
+        * "equal": devide the field into equal parts
+        * an array of explicitly given thresholds
+        Default: "arithmetic"
+    mean : :class:`float`or :any:`None`
+        Mean of the field for "equal" thresholds. Default: np.mean(field)
+    var : :class:`float`or :any:`None`
+        Variance of the field for "equal" thresholds. Default: np.var(field)
+
+    Returns
+    -------
+    :class:`numpy.ndarray`
+        Transformed field.
+    """
+    field = np.asarray(field)
+    if thresholds == "arithmetic":
+        # just in case, sort the values
+        values = np.sort(values)
+        thresholds = (values[1:] + values[:-1]) / 2
+    elif thresholds == "equal":
+        mean = np.mean(field) if mean is None else float(mean)
+        var = np.var(field) if var is None else float(var)
+        values = np.asarray(values)
+        n = len(values)
+        p = np.arange(1, n) / n  # n-1 equal subdivisions of [0, 1]
+        rescale = np.sqrt(var * 2)
+        # use quantile of the normal distribution to get equal ratios
+        thresholds = mean + rescale * erfinv(2 * p - 1)
+    else:
+        if len(values) != len(thresholds) + 1:
+            raise ValueError(
+                "discrete transformation: len(values) != len(thresholds) + 1"
+            )
+        values = np.asarray(values)
+        thresholds = np.asarray(thresholds)
+    # check thresholds
+    if not np.all(thresholds[:-1] < thresholds[1:]):
+        raise ValueError(
+            "discrete transformation: thresholds need to be ascending"
+        )
+    # use a separate result so the intermediate results are not affected
+    result = np.empty_like(field)
+    # handle edge cases
+    result[field <= thresholds[0]] = values[0]
+    result[field > thresholds[-1]] = values[-1]
+    for i, value in enumerate(values[1:-1]):
+        result[
+            np.logical_and(thresholds[i] < field, field <= thresholds[i + 1])
+        ] = value
+    return result
+
+
+def array_boxcox(field, lmbda=1, shift=0):
+    """
+    (Inverse) Box-Cox transformation to denormalize data.
+
+    After this transformation, the again Box-Cox transformed field is normal
+    distributed.
+
+    See: https://en.wikipedia.org/wiki/Power_transform#Box%E2%80%93Cox_transformation
```

Comment:
Would it be silly to add a date when the link was last checked?

Reply:
I don't really want to pollute the docs with such information. I don't think this wiki page will change much in the future ;-)